### ISSUE TITLE
BeeSink module GUI cycles through a subset of bees

### ISSUE THREAD

**Issue by ab9rf** [RS485/LogisticsPipes#610]
At least with both Forestry and ExtraBees install, the GUI for the BeeSink module cycles through only a subset of the registered bee species.  The subset that is available varies but is never the complete set of all bees; in my current environment it does the "hive" species from stock Forestry, the colored bees from the "secondary" branch of Extra Bees, and then back to the start, skipping everything else (including most of the standard Forestry bees).  Everything else is inaccessible.  Previously, I could get to far more of the species, but even then not all of them.  I'm not sure what I changed that caused the list of available bees to change; the last thing that was changed in this configuration was upgrading LogisticPipes from .100 to .105. 

Configurations tested:
- 1.7.10, Forge 1291, Forestry 3.4.0.7, BinniesMods 2.0-pre5, LogisticPipes 0.8.3.105
- 1.7.10, Forge 1291, Forestry 3.4.0.7, BinniesMods 2.0-pre8, LogisticPipes 0.8.3.108

**Comment by SonOfTheStars** [comment]
Ill cehck on this.
It shouldnt be doing that unless forestry vhanged the accessibility of their bees code completely

**Comment by ab9rf** [comment]
I've identified the problem here, and it's Forestry's, not LP's.  This is an issue with the Forestry breeding tracker.  For some reason, after initial player connection, the player's breeding tracker (which is used to determine which species the player "knows about") isn't properly updated until some event causes it to be changed.  So on initial connection to a MP server with Forestry, the player will not "know" about species other than the defaults until the player interacts with a bee in any of the several ways that cause the breeding tracker to update.  Basically, if I throw a bee on the ground, then pick it up (which updates the breeding tracker), the GUI starts working as expected.  (I've only observed this in a server-based world.)

Since this is a Forestry bug, not an LP bug, and it has a fairly silly workaround, this issue can be closed.

**Comment by hron84** [comment]
Any way to fix this from Forestry side? /cc @mezz @CovertJaguar @Vexatos

**Comment by mezz** [comment]
Of course. Can you submit a bug report so the issue doesn't get lost?

**Comment by mezz** [comment]
Of course. Thanks!
I created an issue and linked it above.

**Comment by mezz** [comment]
Fixed an issue with the breeding tracker.

To ensure the client gets the breeding tracker, call this from the server before it opens the container:

``` java
// in a new method in ForestryProxy.java
root.getBreedingTracker(world, player.getGameProfile()).synchToPlayer(player);
```

[Here's an example from Forestry](https://github.com/ForestryMC/ForestryMC/blob/b9b71cf015068ac4d19fee2cd7a0e52dacfe0dc5/src/main/java/forestry/pipes/GuiHandlerPipes.java#L65-L67).

**Comment by hron84** [comment]
@mezz  thanks for your quick reaction.

**Comment by hron84** [comment]
@davboecki @theZorro266 as I guess it will fixed in 0.9? Or we can expect a backport too?

**Comment by bziemons** [comment]
As it is only mod interaction with an older (than stable) version of Forestry, it should be fixed it in 0.9 and 0.8.

**Comment by hron84** [comment]
Cool, thanks. Could you please mark here which build contains a fix when it's done? I think I'm not alone using LP for automating Apiaries and Alvearies.
